runtime/protos/runtime.proto

```diff
@@ -17,502 +17,477 @@ option go_package = "github.com/ServiceWeaver/weaver/runtime/protos";
 
 package runtime;
 
-// EnvelopeMsg holds a message sent by the envelope to the managed weavelet.
+// See https://serviceweaver.dev/blog/deployers.html for an overview of the
+// deployer API.
+//
+// A deployer and a weavelet communicate with each other over a pair of Unix
+// pipes. One pipe runs from the deployer to the weavelet, and the other runs in
+// the opposite direction from the weavelet to the deployer. The deployer and
+// weavelet send EnvelopeMsg and WeaveletMsg protocol buffers over these pipes,
+// respectively. We call the part of a deployer that communicates with a
+// weavelet an envelope, hence the name EnvelopeMsg.
+//
+//                 ┌──────────┐   ________   ┌──────────┐
+//                 │          │<─O________)──│          │
+//                 │ ENVELOPE │   ________   │ WEAVELET │
+//                 │          │──O________)─>│          │
+//                 └──────────┘              └──────────┘
+//
+// When an envelope and weavelet first establish their connection, they perform
+// a *handshake*. The envelope sends an EnvelopeInfo, and the weavelet responds
+// with a WeaveletInfo.
+//
+//                   ENVELOPE                  WEAVELET
+//                   ───┬────                  ───┬────
+//                      │ ──────────────────────> │ EnvelopeInfo
+//                      │ <────────────────────── │ WeaveletInfo
+//                      :                         :
+//
+// After the handshake, the envelope and weavelet communicate freely. There are
+// three communication patterns.
+//
+//     1. The envelope can initiate an RPC against the weavelet.
+//     2. The weavelet can initiate an RPC against the envelope.
+//     3. The weavelet can send unacknowledged messages to the envelope.
```

Comment:
somehow I am still not a big fan of making a distinction between RPC, non-RPC. For me they are all RPCs; 1, 2 are acknowledged or it expects a reply, the third one doesn't.

Reply:
Done. I call everything RPCs now, but clarify that some are unacknowledged.